mesonbuild/compilers.py

```diff
@@ -1053,7 +1053,7 @@ def compiles(self, code, env, extra_args=None, dependencies=None, mode='compile'
     def _links_wrapper(self, code, env, extra_args, dependencies):
         "Shares common code between self.links and self.run"
         args = self._get_compiler_check_args(env, extra_args, dependencies, mode='link')
-        return self.compile(code, args.to_native())
+        return self.compile(code, args)
```

Comment:
Actually, maybe I'm wrong because this will get added as `extra_args` in `self.compile()` which will call `to_native()` on it.